wordpress-mobile/WordPress-Android
# Issue/14128 reader related posts UI cleanup + design update

Task #14128

This PR updates related posts UI.

PR size has increased slightly due to following reverts/ removal after design update:
- Remove follow button logic:  9d3243d...d929d4b, 02884be
- Remove redundant fields and view for `ReaderSimplePost`: 3b8b81f


Light mode | Light mode (layout bounds) | Dark mode
-------------|----------------------------|-----------
![device-2021-03-04-164053](https://user-images.githubusercontent.com/1405144/109956028-2db5d100-7d09-11eb-8706-c33601943978.png) | ![device-2021-03-04-164008_markers](https://user-images.githubusercontent.com/1405144/109956044-327a8500-7d09-11eb-91bb-de7e366bce2a.png) | ![device-2021-03-04-164709](https://user-images.githubusercontent.com/1405144/109956183-60f86000-7d09-11eb-8ada-480739ab7df6.png)

To test:

1. Open a wp.com post on Reader tab.
2. Scroll to bottom of the post details page.
3. Notice that related posts are displayed properly (design ref: I5jpMcL84tYRh3FQJZs0fF-fi-315%3A89).

Merge instruction:

1. Wait for design review. 
2. Remove "Needs Design Review" label.
3. Merge this PR to `issue/14128-reader-related-posts`.

PR submission checklist:
